db/blip_handler.go

```diff
@@ -402,6 +402,9 @@ func (bh *blipHandler) handleChanges(rq *blip.Message) error {
 		if missing == nil {
 			// already have this rev, tell the peer to skip sending it
 			output.Write([]byte("0"))
+			if bh.sgr2PullIgnoreSeqCallback != nil {
+				bh.sgr2PullIgnoreSeqCallback(seqStr(change[0]))
```

Comment:
Why are we doing PullIgnore per-change, but calling PullAddExpected once per changes message?  Any reason not to batch up PullIgnore as well, to minimize locking?  If the idea is to checkpoint midway through a large changes message, it looks like a single expectedSeq is going to break that.